### ISSUE TITLE
RackAttack is rate-limiting ingress relay traffic

### ISSUE THREAD

**Issue by DataDrivenMD** [mastodon/mastodon#23722]
### Steps to reproduce the problem

1. Connect local instance to a high-volume relay, such as relay.universeodon.com
2. Enable RAILS_LOG_LEVEL=debug
3. Go down the street to get a cup of coffee, then check the logs when you return. 

### Expected behaviour

Relay traffic should be exempt from rate-limiting

### Actual behaviour

Relay traffic is not exempt from rate-limiting

### Detailed description

<img width="1314" alt="relay" src="https://user-images.githubusercontent.com/116459476/219933811-077808a9-ee15-471b-8df9-a75f6d8dc255.png">

### Specifications

- Mastodon 4.1.0
- Relay URL: https://relay.universeodon.com/inbox

**Comment by Gargron** [comment]
ActivityPub paths are not subject to rate-limits. However, it seems like you have the relay's IP blocked.

**Comment by DataDrivenMD** [comment]
I just checked, and I don't have the relay IP on my blocklist. I queried Postgres directly, and didn't find the IP address listed in the `ip_blocks` table either. Is there somewhere else I should check?

**Comment by Gargron** [comment]
Never mind, nothing is getting blocked in the screenshot. It says it returned status=202 right there. IP blocks are checked on every request however.

**Comment by DataDrivenMD** [comment]
Right, I see the 202s, but the actual messages aren't coming through, which I now realize that I failed to mention in my original report. So, taking a step back: since upgrading to 4.1.0 I noticed that ingress traffic would be humming along, then suddenly drop-off. I didn't see anything obvious in the logs using the `INFO` log level, so I changed it to `DEBUG`. It was at that point that I noticed that delivery attempts were being made, but that those messages weren't being accepted. In the screenshot, for example, there's an inbound from `TheTweetOfGod@universeodon.com`, whom I follow. I never received that message on the front-end (and hundreds of others from other accounts that I follow).

Upon noticing this, I disabled the relay via the web admin console. After a few hours, the problem seemed to resolve and I began receiving messages at a reasonable cadence.

My working hypothesis is that, my request to block one or more instances that contribute to the relay is getting (mis)interpreted as a request to block all messages that are coming through the relay. Is that even possible? At what point in the stack does Mastodon decide whether to allow a message through? And at what point does it reconcile duplicate messages (i.e. those that arrive via the relay vs. those that arrive directly from the origin server)?
